### DISCUSSION TITLE
Does AvanteEdit for the selection use the whole file as context?

### DISCUSSION THREAD

**Discussion by lenohard** [yetone/avante.nvim#540] [General]
Sometimes the question is local and there is no need for the whole file as contexts. is this supported elsewhere?

**Comment by aarnphm** [comment]
Are there cases when you just want to ask about one segment of the code without context? It seems like this is not within the scope of edit.

Edit should be as is: It will try to edit a selection of code based on given context.

**Comment by lenohard** [comment]
Sometimes I just want to rewrite a method in a large file. The function of this method is simple and independent. eg: process the input and output formatted version of the input. Before using avante, I usually can copy and paste it directly into ChatGPT and get a satisifed result. It's would be more convinient if I can just select and then rewrite it in-place using Avante.

**Comment by nvimts** [comment]
need the same feature to reduce cost

**Reply by nvimts** [reply]
and for a cleaner chat context
